### ISSUE TITLE
Add "Field Defaults" section

### ISSUE THREAD

**Issue by engram-design** [verbb/formie#863]
Here, we would have a section in Formie's settings to, for each field, open the usual modal for the field settings, and save defaults for all new fields created. 

We'll need to figure out where to store them, within project config, or outside. My gut-feeling is within project config so developers are the only ones with access to save it.

**Comment by MoritzLost** [comment]
Looking forward to this one! Any ETA for when this might land? The issue with the inaccessible default for date fields is holding us back a bit. Is there any hook we could use to change the default for new forms programmatically, while still allowing users to change this option?

Otherwise we could just hardcode `useDatePicker` in the templates, though I don't particularly like that then there's an option that doesn't do anything in the backend. AFAIK there isn't a way to hide/disable options that are overwritten in the templates, right?

+1 for storing defaults in the project config!

**Comment by engram-design** [comment]
It'll be tackled after our Formie 3 major update, as we're pretty busy with that, so a few months away. With any luck, and if we have time, we might be able to sneak it into that, but otherwise scheduled for Formie 2.2.0 (still for Craft 4).

I'd say the "easiest" method is to override the Date field template to just not output the date picker, ever. There are other ways like extending the Date field class to add your own defaults, but that's not great for the future, where you'd have to migrate those fields back to the Formie Date field class when we implement this.

**Comment by MoritzLost** [comment]
@engram-design Looking forward to it!

> I'd say the "easiest" method is to override the Date field template to just not output the date picker, ever. There are other ways like extending the Date field class to add your own defaults, but that's not great for the future, where you'd have to migrate those fields back to the Formie Date field class when we implement this.

Thanks, I think we'll go with the template overrides for now, or just provide our preferred defaults with Stencils. For the field defaults, the 'make a wish' feature for me would be the ability to disable selected field options where I want to enforce a specific value. For example, enforcing that the datepicker is off and removing the datepicker option from date fields would be great. Maybe that could be done in the same interface where the defaults are defined.

I noticed there's also some hooks to define config/options, I'll look into whether those can be used for this!

**Comment by engram-design** [comment]
Yeah, good call on being able to disable the setting being editable as well, I can work that in.

**Comment by svondervoort** [comment]
It would also be nice to have integrations automatically be enabled for new forms created.
For example, we've configured several Catpcha integrations we would like to be active for all forms.

**Comment by engram-design** [comment]
@svondervoort That's how captchas currently work, but they only affect forms created after you've globally enabled the captcha.

**Comment by svondervoort** [comment]
> @svondervoort That's how captchas currently work, but they only affect forms created after you've globally enabled the captcha.

Maybe I'm missing some configuration but I have activated Recaptcha, Duplicate, Honeypot and Javascript Captchas in the Formie Settings. Then when I create a new form and save it the integrations are not enabled for the new form.

**Comment by engram-design** [comment]
@svondervoort That's a bug. Fixed for the next release. To get this early, run `composer require verbb/formie:"dev-craft-5 as 3.0.14"`.

**Comment by svondervoort** [comment]
Any status update on this matter? One of our top priority `defaults` would be the `data retention` configuration.

**Comment by engram-design** [comment]
No updates yet, sorry. 

We're almost done with 2.2, and will be moving on to Formie 2.3 where this is scheduled for soon.

**Comment by svondervoort** [comment]
And what Formie 3.x release will include these features?

**Comment by engram-design** [comment]
We've marked this to do in the 2.4 milestone (equating to 3.3), but we'll look at tackling this sooner in 2.3.

**Comment by bfopma** [comment]
@engram-design similar to @svondervoort, the ability to set a global default for the `data_retention` configuration is of high priority for us. The default setting to retain data 'Forever' is problematic for several reasons (legal risk, database size growth, etc.). This one seems like it belongs on the plugin's 'Settings' section as a value that can be set.

**Comment by bartdigitalpulse** [comment]
Would like to see a global/default data retention setting as well.

**Comment by engram-design** [comment]
Added in [4.0.0-beta.3](https://github.com/verbb/formie/releases/tag/4.0.0-beta.3).
